app/models/host_info_providers/puppet_info.rb

```diff
@@ -21,8 +21,23 @@ def puppetclass_parameters
       values = keys.values_hash(host)
 
       klasses = {}
+      klass_error = {}
       host.classes.each do |klass|
-        klasses[klass.name] = smart_class_params_for(klass, key_hash, values)
+        begin
+          klasses[klass.name] = smart_class_params_for(klass, key_hash, values)
+        rescue StandardError => exc
+          klass_error[klass] = exc
+        end
+      end
+      if klass_error.size
+        # FIXME: Is logger in scope under another name?
+        logger = Foreman::Logging.logger('app')
+        klass_error.sort.each do |klass, exc|
+          logger.error("Smart class parameter rendering of `#{klass.name}` failed on #{host.name}: #{exc}")
+        end
+        klass_names = klass_error.keys.map { |klass| klass.name }
+        # FIXME: RuntimeExceptions are lame. Custom Exception object with the klass_error data for HTML rendering?
+        raise "Smart class parameter rendering failed for #{klass_names}"
```

Comment:
Please use `Foreman::Exception` for that. https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions